File: meme_share_react_native/src/service/api_client.ts
import axios from 'axios';

export const api = axios.create({
    baseURL: 'https://alperensaracdeneme.com/meme/',
    timeout: 30000,
    headers: {
        Accept: 'application/json',
    },
});

// PHP bazen text/plain döndürebilir; axios çoğu zaman yine parse eder.
// Gerekirse burada interceptor eklenebilir.

File: meme_share_flutter/lib/service/api_client.dart
import 'package:dio/dio.dart';

class ApiClient {
  static const String baseUrl = 'https://alperensaracdeneme.com/meme/';

  static Dio createDio() {
    final dio = Dio(
      BaseOptions(
        baseUrl: baseUrl,
        connectTimeout: const Duration(seconds: 20),
        receiveTimeout: const Duration(seconds: 30),
        sendTimeout: const Duration(seconds: 60),
        headers: {
          'Accept': 'application/json',
        },
        // PHP bazen text/plain dönebiliyor. Dio yine decode eder ama
        // gerekirse responseType ayarlanabilir.
      ),
    );

    // İstersen debug log interceptor ekleyebilirsin:
    // dio.interceptors.add(LogInterceptor(
    //   requestBody: true,
    //   responseBody: true,
    // ));

    return dio;
  }
}
